### PR TITLE
fix: handle null response.output in parse_response

### PR DESCRIPTION
**Pull request by nightcityblade**
## Summary

Handle responses where `response.output` is `null` when parsing response snapshots, avoiding a `TypeError` during `response.completed` handling.

## Changes

- Iterate over `response.output or []` in `parse_response()`
- Add a targeted regression test covering a `Response` with null output

## Testing

- `uv run pytest tests/lib/responses/test_responses.py -q`
- `uv run pyright src/openai/lib/_parsing/_responses.py tests/lib/responses/test_responses.py`

Fixes openai/openai-python#3325

### CODE REVIEW

File: src/openai/lib/_parsing/_responses.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def parse_response(
 ) -> ParsedResponse[TextFormatT]:
     output_list: List[ParsedResponseOutputItem[TextFormatT]] = []
 
-    for output in response.output:
+    for output in response.output or []:
```
[the remark this answers is not in the document]
Comment (nightcityblade) [reply src/openai/lib/_parsing/_responses.py:61]: Thanks for catching this. I updated the stream accumulator so that when the terminal `response.completed` payload has `output: null`, it preserves the accumulated snapshot output while keeping the final response metadata. I also added a regression test covering streamed output deltas followed by a completed event with null output.

Tests run:
- `uv run pytest tests/lib/responses/test_responses.py -q`
- `uv run ruff check src/openai/lib/streaming/responses/_responses.py tests/lib/responses/test_responses.py`
- `uv run ruff format --check src/openai/lib/streaming/responses/_responses.py tests/lib/responses/test_responses.py`

**Comment by marcuswood-oai** [comment]
Thank you for the contribution! The null-output parsing and stream recovery fix has landed in #3345, so I am closing this PR as superseded. We appreciate your help!
